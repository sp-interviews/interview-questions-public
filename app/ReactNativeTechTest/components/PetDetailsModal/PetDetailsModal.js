import React from 'react';
import PropTypes from 'prop-types';
import {StyleSheet, Text, View, TouchableHighlight} from 'react-native';
import {Colors} from '../../constants/Colors';
import ImageCircle from '../ImageCircle/ImageCircle';
import Button from '../Button/Button';

const PetDetailsModal = ({pet, onPressCloseModal, onPressDelete}) => {
  if (!pet) {
    return null;
  }
  const {image, name, type, feeds} = pet;

  return (
    <TouchableHighlight
      style={styles.container}
      onPress={onPressCloseModal}
      underlayColor="transparent">
      <View style={styles.modal}>
        <View style={styles.imageContainer}>
          <ImageCircle image={image} size="large" />
        </View>
        <Text style={styles.name}>{name}</Text>
        <Text style={styles.detail}>{type}</Text>
        <Text
          style={styles.detail}>{`Needs feeding ${feeds} times a day`}</Text>
        <View style={styles.buttonContainer}>
          <Button
            label="Close"
            backgroundColor={Colors.blue}
            onPress={onPressCloseModal}
          />
          <Button
            label="Delete"
            backgroundColor={Colors.red}
            onPress={onPressDelete}
          />
        </View>
      </View>
    </TouchableHighlight>
  );
};

PetDetailsModal.propTypes = {
  pet: PropTypes.object,
  onPressCloseModal: PropTypes.func,
  onPressDelete: PropTypes.func,
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  modal: {
    backgroundColor: Colors.white,
    paddingHorizontal: 30,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 30,
    shadowColor: Colors.black,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  name: {
    fontSize: 20,
    textAlign: 'center',
    marginBottom: 8,
  },
  detail: {
    fontSize: 14,
    textAlign: 'center',
    marginVertical: 8,
  },
  buttonContainer: {
    marginTop: 25,
  },
  imageContainer: {
    marginTop: -40,
    marginBottom: 20,
  },
});

export default PetDetailsModal;
