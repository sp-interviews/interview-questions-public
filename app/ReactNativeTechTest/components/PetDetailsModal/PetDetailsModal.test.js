/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import PetDetailsModal from './PetDetailsModal';

describe('Components - Header', () => {
  it('renders', () => {
    const petDetails = {
      name: 'Bob',
      type: 'Cat',
      feeds: 2,
    };
    const tree = renderer.create(<PetDetailsModal pet={petDetails} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('When no type is provided', () => {
    it('renders without type', () => {
      const petDetails = {
        name: 'Bob',
        feeds: 2,
      };
      const tree = renderer
        .create(<PetDetailsModal pet={petDetails} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
