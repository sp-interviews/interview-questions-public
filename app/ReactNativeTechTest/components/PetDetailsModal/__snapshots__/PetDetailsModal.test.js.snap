// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - Header When no type is provided renders without type 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderRadius": 30,
        "elevation": 5,
        "justifyContent": "center",
        "paddingHorizontal": 30,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 20,
          "marginTop": -40,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#444",
            },
            Object {
              "borderRadius": 80,
              "height": 80,
              "width": 80,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "https://reactnative.dev/img/tiny_logo.png",
            }
          }
          style={
            Object {
              "borderRadius": 80,
              "height": 80,
              "width": 80,
            }
          }
        />
      </View>
    </View>
    <Text
      style={
        Object {
          "fontSize": 20,
          "marginBottom": 8,
          "textAlign": "center",
        }
      }
    >
      Bob
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
          "marginVertical": 8,
          "textAlign": "center",
        }
      }
    >
      Needs feeding 2 times a day
    </Text>
    <View
      style={
        Object {
          "marginTop": 25,
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "blue",
            "borderRadius": 45,
            "marginBottom": 15,
            "opacity": 1,
            "paddingHorizontal": 15,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFF",
              "textAlign": "center",
            }
          }
        >
          Close
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "red",
            "borderRadius": 45,
            "marginBottom": 15,
            "opacity": 1,
            "paddingHorizontal": 15,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFF",
              "textAlign": "center",
            }
          }
        >
          Delete
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Components - Header renders 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderRadius": 30,
        "elevation": 5,
        "justifyContent": "center",
        "paddingHorizontal": 30,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 20,
          "marginTop": -40,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#444",
            },
            Object {
              "borderRadius": 80,
              "height": 80,
              "width": 80,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "https://reactnative.dev/img/tiny_logo.png",
            }
          }
          style={
            Object {
              "borderRadius": 80,
              "height": 80,
              "width": 80,
            }
          }
        />
      </View>
    </View>
    <Text
      style={
        Object {
          "fontSize": 20,
          "marginBottom": 8,
          "textAlign": "center",
        }
      }
    >
      Bob
    </Text>
    <Text
      id="type"
      style={
        Object {
          "fontSize": 14,
          "marginVertical": 8,
          "textAlign": "center",
        }
      }
    >
      Cat
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
          "marginVertical": 8,
          "textAlign": "center",
        }
      }
    >
      Needs feeding 2 times a day
    </Text>
    <View
      style={
        Object {
          "marginTop": 25,
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "blue",
            "borderRadius": 45,
            "marginBottom": 15,
            "opacity": 1,
            "paddingHorizontal": 15,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFF",
              "textAlign": "center",
            }
          }
        >
          Close
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "red",
            "borderRadius": 45,
            "marginBottom": 15,
            "opacity": 1,
            "paddingHorizontal": 15,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFF",
              "textAlign": "center",
            }
          }
        >
          Delete
        </Text>
      </View>
    </View>
  </View>
</View>
`;
